workflow.Rmd

```diff
@@ -0,0 +1,473 @@
+---
+title: "PACTA_analysis workflow"
+author: "Clare"
+date: "08/02/2021"
+output: html_document
+---
+
+## Aim
+
+The aim of this document is to outline the workflow for running a PACTA project. 
+
+## PACTA_analysis Repo
+
+The PACTA_analysis repo contains the primary PACTA functions required to run a portfolio alignment assessment for investors. The current repository is built in a way that the functions work for both an online application - ie for the P2020 projects and platform; as well as the original application which is running a project or generating results offline.
+
+Whether online or offline, running an analysis has three components
+
+1. Portfolio Cleaning: Clean the input portfolio and merge in financial data, to categorise each holding and identify whether it's equity (EQ) or corporate bonds (CB)
+2. Run Analysis: Merge in ald + scenarios, and group results at company, portfolio and regional level
+3. Results: Present the results in a clear output format
+
+Both the online and offline scripts can be used to generate reports offline. The difference is that the offline set of scripts is set up to allow the generation of results for multiple portfolios and investors in one run, whereas online is focussed on quickly generating a report for one portfolio at once. While all steps maintain the grouping by investor and portfolio, to allow for this, it hasn't been optimised for running multiple portflios as once. The online tool however has been designed for the interactive reports, whereas the offline scripts have primarily been used to generate the static pdf results.  
+
+
+### Portfolio Cleaning
+
+This occurs in web_tool_script_1.R (online) or 2_project_input_analysis.R (offline). 
+
+Both start with reading in the necessary files. This includes some file cleaning, that should be shifting to the data_preprataion repository.  
+
+## Difference between online and offline
+
+There are two sets of scripts that call the functions required for both offline or online functionality. 
+
+Online: The "web_tool_script" set from 1 to 3 runs through the components of the analysis
+Offline: The 1,2,3,4 scripts run similarly, however have an initialisation script that should be run before each online in order to set up project pathways and naming. This allows for added flexibility as you may online want to be focussing on one of the components of the analysis. Ie you've already generated results and just want to create output files. 
+
+
+# Running a project
+
+
+## Offline 
+This steps through the functions required to run a project using the original scripts. The aim of this script is to define project paths and set parameters used in the following scripts. 
+
+### Set up parameters and paths
+```{r 1_portfolio_check_initialisation, echo = FALSE}
+
+devtools::load_all()
```

Comment:
This might be too much detail for now. I would start by documenting the workflows at the highest level, showing clearly what the inputs and outputs are, and making sure this rmarkdown document acutally runs in anyone's computer. Once we have that we could move deeper into details if necessary.